numba/tests/test_unicode.py

```diff
@@ -337,6 +349,41 @@ def test_find(self, flags=no_pyobj_flags):
                                  cfunc(a, substr),
                                  "'%s'.find('%s')?" % (a, substr))
 
+    def test_count(self):
+        pyfunc = count_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES + ['aaaa']:
+            extras = ['', ' ', 'xx', s[::-1], s[:-2], s[3:], s, s + s, 'aa']
+            for sub in [x for x in extras]:
+                self.assertEqual(pyfunc(s, sub),
+                                 cfunc(s, sub),
+                                 "'%s' in '%s'?" % (sub, s))
+
+    def test_count_with_start_end(self):
+        pyfunc = count_with_start_end_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES:
+            extras = ['', ' ', 'xx', s[::-1], s[:-2], s[3:], s, s + s]
+            for sub in [x for x in extras]:
+                for i , j in zip(range(-2,4), (0,6)):
```

Comment:
Yes that is intended because I wanted to cover all kinds of cases in the test. I had thought about using itertools but I can't quite remember now why I ended up using zip. Does itertool have significant benefit over zip?

Reply:
If the requirement is to just testing two values then perhaps just spell those out instead of using range? On the basis that it seems like testing more values would be useful, consider the following:
```python
In [51]: for i, j in zip(range(-2,4), (0,6)): 
    ...:     print(i, j) 
    ...:                                                                                                                        
-2 0
-1 6

In [52]: for i, j in ((-2, 0), (-1, 6)): 
    ...:     print(i, j) 
    ...:                                                                                                                        
-2 0
-1 6

In [53]: for i, j in itertools.product(range(-3, 3), range(-2, 2)): 
    ...:     print(i, j) 
    ...:                                                                                                                        
-3 -2
-3 -1
-3 0
-3 1
-2 -2
-2 -1
-2 0
-2 1
-1 -2
-1 -1
-1 0
-1 1
0 -2
0 -1
0 0
0 1
1 -2
1 -1
1 0
1 1
2 -2
2 -1
2 0
2 1

```

Reply:
Ooh I realize my mistake now! Sorry I had some misunderstanding. I hadn't realized until now that I should have used "range" in front of (0,6). I get what you're saying now. I'll fix it. Thanks!

Reply:
I think that if you use two zipped ranges then the result will be monotonic.
```pycon
In [56]: for i , j in zip(range(-2,4), range(0,6)): 
    ...:     print(i, j) 
    ...:      
    ...:                                                                                                                        
-2 0
-1 1
0 2
1 3
2 4
3 5
```

 whereas `itertools.product` will cover the space `start > end, start == end, start < end` with a variety of negative and positive values.

Reply:
Yes you're right! Using itertools.product will cover good range. I will change that.